### PR TITLE
[Merged by Bors] - feat: port the basic lemmas in Init.Meta.WellFoundedTactics

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
This file contained a few lemmas that should really have gone elsewhere already, before the actual meta code. I'm only porting those in this PR.

### CODE REVIEW

File: Mathlib/Init/Meta/WellFoundedTactics.lean [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+protected def PSum.Alt.sizeOf.{u, v} {α : Type u} {β : Type v} [SizeOf α] [SizeOf β] : PSum α β → ℕ
+  | PSum.inl a => SizeOf.sizeOf a
+  | PSum.inr b => SizeOf.sizeOf b
+#align psum.alt.sizeof PSum.Alt.sizeOf
+
+@[reducible]
+protected def PSum.sizeOfAlt.{u, v} (α : Type u) (β : Type v) [SizeOf α] [SizeOf β] :
+    SizeOf (PSum α β) :=
+  ⟨PSum.Alt.sizeOf⟩
+#align psum.has_sizeof_alt PSum.sizeOfAlt
```
Comment (gebner) [inline Mathlib/Init/Meta/WellFoundedTactics.lean:48]:
```suggestion
```

I think this falls under Lean 3 implementation detail (it's only referenced from the WF recursion module).

File: Mathlib/Init/Meta/WellFoundedTactics.lean [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+prelude
+import Mathlib.Init.Data.Nat.Lemmas
+
+-- Porting note: extremely partial port!
+
+-- TODO(Leo): move this lemma, or delete it after we add algebraic normalizer.
+-- TODO(Leo): move this lemma, or delete it after we add algebraic normalizer.
```
Comment (gebner) [inline Mathlib/Init/Meta/WellFoundedTactics.lean:17]:
```suggestion
```

I think it's time to retire the comment.

File: Mathlib/Init/Meta/WellFoundedTactics.lean [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+! This file was ported from Lean 3 source module init.meta.well_founded_tactics
+! leanprover-community/lean commit 855e5b74e3a52a40552e8f067169d747d48743fd
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+prelude
+import Mathlib.Init.Data.Nat.Lemmas
+
+-- Porting note: extremely partial port!
```
Comment (gebner) [inline Mathlib/Init/Meta/WellFoundedTactics.lean:14]:
```suggestion
-- Porting note: meta code used to implement well-founded recursion is not ported
```
As it should be.

File: Mathlib/Init/Meta/WellFoundedTactics.lean (hunk 2 above)
Comment (hrmacbeth) [reply Mathlib/Init/Meta/WellFoundedTactics.lean]: Probably Gabriel meant for you also to delete the similar comments on the other lemmas?

File: Mathlib/Init/Meta/WellFoundedTactics.lean [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    assumption
+  Nat.zero_lt_succ _
+#align nat.zero_lt_one_add Nat.zero_lt_one_add
+
+#align nat.lt_add_right Nat.lt_add_right
+
+theorem Nat.lt_add_left (a b c : Nat) : a < b → a < c + b := fun h =>
+  lt_of_lt_of_le h (Nat.le_add_left _ _)
+#align nat.lt_add_left Nat.lt_add_left
+
```
Comment (jcommelin) [inline Mathlib/Init/Meta/WellFoundedTactics.lean:34]:
```suggestion

/-
The remainder of the original Lean 3 source module is subsumed by Lean 4 core.
-/

```

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**